### DISCUSSION TITLE
Various descriptions of the error when clicking on a broken link in Firefox in Selenide and Selenium

### DISCUSSION THREAD

**Discussion by prafair** [selenide/selenide#2724] [Q&A] [answered]
In Firefox when clicking on a non-working link (for example, https://example), Selenide displays an error:

> Element not found {#link}
> Expected: clickable: interactable and enabled

But in Selenium, when performing similar actions, another error is displayed:

> org.openqa.selenium.WebDriverException: Reached error page: about:neterror?e=dnsNotFound&u=https%3A//example/&c=UTF-8&d=We%20can%E2%80%99t%20connect%20to%20the%20server %20at%20example.

In Chrome and Safari these tests are green.

My question is where is the correct error: in Selenide or Selenium? It looks like there is a more accurate error in Selenium.

And maybe in Chrome and Safari the test should fail, because according to the standard W3C, following a link should also be checked and the response status should be returned?

---- 
Minimum code to reproduce:
```
import com.codeborne.selenide.Configuration;
import com.codeborne.selenide.junit5.TextReportExtension;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;
import org.openqa.selenium.By;

import static com.codeborne.selenide.Selenide.*;

@ExtendWith(TextReportExtension.class)
class ClickNavigateTest {

    @ParameterizedTest
    @ValueSource(strings = {"chrome", "firefox", "safari"})
    void seleniumClickNavigateTest(String browser) {
        closeWebDriver();
        Configuration.browser = browser;
        open("/your_html_file.html");
        webdriver().object().findElement(By.id("link")).click();
    }

    @ParameterizedTest
    @ValueSource(strings = {"chrome", "firefox", "safari"})
    void selenideClickNavigateTest(String browser) {
        closeWebDriver();
        Configuration.browser = browser;
        open("/your_html_file.html");
        $("#link").click();
    }
}
```

HTML:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
<a href="https://example" id="link">Link</a>
</body>
</html>
```

Versions:

> Selenide 7.2.3
> JUnit 5.9.1

**Comment by BorisOsipov** [comment]
>Selenide displays an error:
>Element not found {#link}
>Expected: clickable: interactable and enabled

Please read the error. It says "Expected: clickable: interactable and enabled". Selenide is **trying** to click and can't because the conditions interactable and enabled are not met. Selenide doesn't even click. It is incorrect to compare two cases where the test clicks and does not click.

>And maybe in Chrome and Safari the test should fail, because according to the standard W3C, following a link should also be checked and the response status should be returned?

It is browser driver implementation specific and outside the scope of Selenide.

**Reply by prafair** [reply]
But the link is clickable and Selenide successfully clicks and we go to the broken page:

<img width="1016" alt="Снимок экрана 2024-04-24 в 13 33 36" src="https://github.com/selenide/selenide/assets/15040493/89f2da33-223c-40e3-b2e1-790f8d936769">

**Reply by BorisOsipov** [reply]
From your screen I see the same failed click with the same reason.The full stacktrace may give some options to investigate, but there is only a report screen.
I have tried your repro code and `seleniumClickNavigateTest\selenideClickNavigateTest` are always green for Chrome\Firefox.

**Reply by prafair** [reply]
https://pastebin.com/0FEnA98C

This is strange, it didn’t work for me in Firefox. I guess only if there is some version/platform dependency

**Reply by BorisOsipov** [reply]
I guess the problem is that Selenide underhood can retry failed clicks with `WebDriverException: Reached error page`. Obviously, on a broken page, there is no #link, and you receive the wrong last error

**Reply by prafair** [reply]
Aah, yes, it seems that this is happening. Can this be considered a feature or a bug? ;))

**Reply by BorisOsipov** [reply, marked as answer]
It is the bug.

**Reply by prafair** [reply]
OK, thanks for the answers! I see that you have already created a bug) Nice)
